### DISCUSSION TITLE
Any method to facilitate communication between worskspaces/windows/instances of vscode

### DISCUSSION THREAD

**Discussion by jakereifer** [microsoft/vscode-discussions#1022] [Extension Development QnA] [answered]
If my extension is installed and I have two different windows (workspaces or instances) of vs code open. Is there any way to have communication between the two instances in the extension code.

For instance, if I call a command in instance 1, will it have side effects in both windows or just the instance 1? Or if I log out in instance 1, will it show to be logged out in the instance 2's session menu or just instance 1's?

It seems that both the node native `EventEmitter` and vscode's `EventEmitter<T>` both do not cross the workspace boundary. Is there any other part of the vscode api that can be used to accomplish reacting to events across this boundary?

**Comment by isidorn** [comment]
@bpasero might have good suggestions here

**Comment by bpasero** [comment, marked as answer]
I am not aware of a way to communicate from the extension in the one window to the extension in the other window, at least not using any of the VS Code APIs.
